rustls/src/stream.rs

```diff
@@ -80,17 +72,9 @@ where
         // We call complete_io() in a loop since a single call may read only
         // a partial packet from the underlying transport. A full packet is
         // needed to get more plaintext, which we must do if EOF has not been
-        // hit. Otherwise, we will prematurely signal EOF by returning without
-        // writing anything. We determine if EOF has actually been hit by
-        // checking if 0 bytes were read from the underlying transport.
+        // hit.
         while self.conn.wants_read() {
-            let at_eof = self.conn.complete_io(self.sock)?.0 == 0;
-            if at_eof {
-                if let Ok(io_state) = self.conn.process_new_packets() {
-                    if at_eof && io_state.plaintext_bytes_to_read() == 0 {
-                        return Ok(());
-                    }
-                }
+            if self.conn.complete_io(self.sock)?.0 == 0 {
```

Comment:
Looks like codecov/patch fails because this line was added and not tested, however, just wanted to mention that this function was also not being tested before this PR*, so in theory coverage is not being degraded. Quite the opposite, by removing code that was not being tested before, the overall coverage in the codebase is increasing.

Coverage before this PR:
![image](https://github.com/rustls/rustls/assets/47480384/535a0c11-dffd-494f-83ef-28d4e6352431)

Reply:
Thanks for taking a look at the diff :-)

It seemed easy to add some coverage for `read_buf` so I went ahead and did that in my last force push. The coverage check is happy now with 100% of the diff covered and the overall coverage up +0.23%.

On a related note, the Stream/StreamOwned tests (both the ones that existed before, and the ones that I added) are pretty duplicative. I spent an hour or so trying to wrangle some commonality out of these but ultimately found that the generics/trait contortions required to make it work made the tests harder to follow, so I matched existing style and left the duplication.

Reply:
> I might take another crack at de-duplicating in a follow-up :thinking: 

Couldn't resist trying: https://github.com/rustls/rustls/pull/1306 I couldn't remove all of the duplication but it's a bit better :stars: